metadb/service.go

```diff
@@ -0,0 +1,31 @@
+package metadb
+
+import (
+	"context"
+
+	"github.com/pastelnetwork/gonode/metadb/rqlite/store"
+)
+
+const (
+	logPrefix = "rqlite"
+)
+
+// Service represents the rqlite cluster
+type Service struct {
+	config *Config       // the service configuration
+	db     *store.Store  // the store for accessing the rqlite cluster
+	ready  chan struct{} // mark the rqlite node is started
+}
+
+// NewService returns a new service for rqlite cluster
+func NewService(config *Config) *Service {
+	return &Service{
+		config: config,
+		ready:  make(chan struct{}, 1),
+	}
+}
```

Comment:
```suggestion
// Service represents the rqlite cluster
type Service interface {
	....
}

type service struct {
	config *Config       // the service configuration
	db     *store.Store  // the store for accessing the rqlite cluster
	ready  chan struct{} // mark the rqlite node is started
}

// NewService returns a new service for rqlite cluster
func New(config *Config) Service {
	return &service{
		config: config,
		ready:  make(chan struct{}, 1),
	}
}
```

When we write unit test for business logic services, we use mock, rather than really DB

Reply:
done